tap_zendesk/client.py

```diff
@@ -25,29 +30,31 @@
 class ZendeskStream(RESTStream):
     """Zendesk stream class."""
 
+    def __init__(self, tap, name=None, schema=None, path=None):
+        super().__init__(tap, name, schema, path)
+        self.last_request_time = 0  # Initialize last_request_time
+        self.allow_redirects = False
+
     @property
     def url_base(self) -> str:
         """Return the API URL root, configurable via tap settings."""
-        # TODO: hardcode a value here, or retrieve it from self.config
-        return "https://api.mysample.com"
-
-    records_jsonpath = "$[*]"  # Or override `parse_response`.
-
-    # Set this value or override `get_new_paginator`.
-    next_page_token_jsonpath = "$.next_page"  # noqa: S105
+        subdomain = self.config.get("subdomain", "")
+        self.logger.debug(f'subdomain: {subdomain}')
+        return f"https://{subdomain}.zendesk.com"
 
     @property
-    def authenticator(self) -> APIKeyAuthenticator:
+    def authenticator(self) -> BasicAuthenticator:
         """Return a new authenticator object.
 
         Returns:
             An authenticator instance.
         """
-        return APIKeyAuthenticator.create_for_stream(
+        email = self.config.get("email", "")
+        api_token = self.config.get("api_token", "")
+        return BasicAuthenticator.create_for_stream(
             self,
-            key="x-api-key",
-            value=self.config.get("auth_token", ""),
-            location="header",
+            username=f"{email}/token",
+            password=api_token
         )
 
     @property
```

Comment:
If we are not changing the http_headers, we can remove it since its already implemented in `RESTStream`.